manager/integration/tests/test_csi_snapshotter.py

```diff
@@ -674,23 +732,213 @@ def test_csi_snapshot_with_snap_param(client, volume_name): # NOQA
+                                  apps_api, # NOQA
+                                  client, # NOQA
+                                  make_deployment_with_pvc, # NOQA
+                                  volumesnapshotclass, # NOQA
+                                  core_api # NOQA
+                                 )
+
+    pvc_name = vol.name + "-pvc"
+    deployment_name = deployment['metadata']['name']
+    csivolsnap = volumesnapshot(vol.name + "-volumesnapshot",
+                                "default",
+                                csisnapclass["metadata"]["name"],
+                                "persistentVolumeClaimName",
+                                pvc_name)
+
+    wait_for_volumesnapshot_ready(
+                            volumesnapshot_name=csivolsnap["metadata"]["name"],
+                            namespace='default',
+                            ready_to_use=True)
+
+    # Step 1 Test create new volume from CSI snapshot
+    # Source volume is attached && Longhorn snapshot exist
+    pvc['spec']['storageClassName'] = 'longhorn'
+    pvc['spec']['dataSource'] = {
+        'name': csivolsnap["metadata"]["name"],
+        'kind': 'VolumeSnapshot',
+        'apiGroup': 'snapshot.storage.k8s.io'
+    }
+    pvc['spec']['resources']['requests']['storage'] = str(500 * Mi)
+    create_pvc(pvc)
+
+    pv_name = wait_and_get_pv_for_pvc(core_api,
+                                      pvc['metadata']['name']).metadata.name
+    new_deployment_name = pv_name + "-dep"
+    new_deployment = make_deployment_with_pvc(new_deployment_name,
+                                              pvc['metadata']['name'])
+    new_deployment["spec"]["replicas"] = 1
+    apps_api.create_namespaced_deployment(body=new_deployment,
+                                          namespace='default')
+
+    common.wait_for_volume_status(client, pv_name,
+                                  common.VOLUME_FIELD_STATE,
+                                  common.VOLUME_STATE_ATTACHED)
+    data_path = "/data/test"
+    pod = common.wait_and_get_any_deployment_pod(core_api, new_deployment_name)
+    created_md5sum = get_pod_data_md5sum(core_api, pod.metadata.name,
+                                         data_path)
+
+    assert expected_md5sum == created_md5sum
+
+    # Source volume is detached
+    deployment["spec"]["replicas"] = 0
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment_name)
+    common.wait_for_volume_status(client, vol.name,
+                                  common.VOLUME_FIELD_STATE,
+                                  common.VOLUME_STATE_DETACHED)
+
+    new_pvc1 = pvc
+    new_pvc1['metadata']['name'] = pvc['metadata']['name'] + "new-pvc1"
+    create_pvc(new_pvc1)
+    check_pvc_in_specific_status(core_api,
+                                 new_pvc1['metadata']['name'], "Pending")
```

Comment:
This could be a false positive because initially every PVC is in the `pending` state. We need to wait for a while and confirm that the PVC is still in pending state. One way to achieve this by modifying the function `check_pvc_in_specific_status` to have similar logic to function https://github.com/longhorn/longhorn-tests/blob/5f89bafbb6acbcd1ace11c1d468f2f7cd95573ff/manager/integration/tests/test_csi_snapshotter.py#L287-L306

Reply:
I modified coed for review, thank  you

Reply:
I think this function still have a bit problem:
```go
def check_pvc_in_specific_status(api, pvc_name, status):
    for i in range(RETRY_EXEC_COUNTS):
        pvcs = api.list_namespaced_persistent_volume_claim("default")
        for item in pvcs.items:
            if item.metadata.name == pvc_name:
                break
        time.sleep(RETRY_INTERVAL)

    assert item.status.phase == status
```
Let say we create a new PVC, initially the state of the PVC is "pending". The above function check and return immediately that the PVC is in pending. That is not what we want. We want to make sure that the PVC remaining in pending for a period of time. Therefore, how do you think about this:
```go
def check_pvc_in_specific_status(api, pvc_name, status):
    for i in range(RETRY_EXEC_COUNTS):
        claim = api.read_namespaced_persistent_volume_claim(name=pvc_name, namespace='default')
        if claim.status.phase == "bound":
            break
        time.sleep(RETRY_INTERVAL)

    assert item.status.phase == status
```

Reply:
HI @PhanLe1010.

Reply:
Thank you for the fix!